src/ngx_rewrite_driver_factory.cc

```diff
@@ -151,6 +151,10 @@ RewriteOptions* NgxRewriteDriverFactory::NewRewriteOptions() {
   return options;
 }
 
+RewriteOptions* NgxRewriteDriverFactory::NewRewriteOptionsForQuery() {
+  return new NgxRewriteOptions(thread_system());
```

Comment:
> That may also be contributing to this problem. Does it work for you at least to change to setting the default rewrite-level instead of setting the rewrite level?

I had also expected that to work, but it doesn't.  Setting the default rewrite level to core filters doesn't enable core filters.

I'm not very happy with my fix as is, but:
* I've put a lot of time into trying to figure out what the problem is, and not finding it
* The fix I have does fix the only place we've seen this bug, and it makes sense that it would fix this bug.